cpp/src/jit/parser.cpp

```diff
@@ -387,52 +326,34 @@ std::string ptx_parser::parse()
 
   input_arg_list.clear();
   // Go directly to the .func mark
-  const size_t length = no_comments.size();
-  size_t start        = no_comments.find(".func") + 5;
-  if (start == length + 5) {
+  size_t offset = no_comments.find(".func");
```

Comment:
Could maybe use `search`, but it's much wordier. It would be nice if`string::find` returned iterators instead of `size_t`.
```cpp
auto func = std::string(".func");
auto f = std::search(no_comments.cbegin(), no_comments.cend(), func.cbegin(), func.cend()) + func.size();
if (f >= no_comments.cend()) { /* bad stuff */ }
```
```

Reply:
I like using `std::search`, it is more consistent with using the generic `std::find` instead of the `find` method on string. I think I actually thought about changing it but didn't think there was an equivalent in the STL. `std::search` didn't come to mind. Thanks!

Reply:
Done :)